### ISSUE TITLE
Thanks for your Fix，But I still get warning from Chrome & Firefox without 'secure'

### ISSUE THREAD

**Issue by mugua** [Veriteworks/CookieFix#7]
Hi, Thanks for your Fix，But I still get warning from Chrome & Firefox without 'secure'。
The Chrome warning is below:
A cookie associated with a cross-site resource at http://baidu.com/ was set without the `SameSite` attribute. A future release of Chrome will only deliver cookies with cross-site requests if they are set with `SameSite=None` and `Secure`. You can review cookies in developer tools under Application>Storage>Cookies and see more details at https://www.chromestatus.com/feature/5088147346030592 and https://www.chromestatus.com/feature/5633521622188032.
The Firefox warning is below:
由于 Cookie “PHPSESSID”的“sameSite”属性设置为“none”，但缺少“secure”属性，此 Cookie 未来将被拒绝。若要了解“sameSite“的更多信息，请参阅：
(This cookie will be rejected in the future because the 'samesite' property of the cookie 'phpsessid' is set to 'none', but the 'secure' property is missing. To learn more about "samesite," see:)
https://developer.mozilla.org/docs/Web/HTTP/Cookies

**Comment by HirokazuNishi** [comment]
Hi @mugua, could you attach your screenshot?

**Comment by mugua** [comment]
@HirokazuNishi  FYI.
![image](https://user-images.githubusercontent.com/15046097/82301187-76a2d000-99ea-11ea-89bb-5733d997f705.png)

**Comment by HirokazuNishi** [comment]
@mugua Thanks. Is your website correctly configured to use SSL/TLS?

**Comment by mugua** [comment]
Yes, it's configured to use SSL/TLS yet.
![image](https://user-images.githubusercontent.com/15046097/82303172-2c6f1e00-99ed-11ea-8fd0-221fd37f1072.png)
![image](https://user-images.githubusercontent.com/15046097/82303189-33962c00-99ed-11ea-898a-964334ce6c74.png)

**Comment by mugua** [comment]
My Nginx setting is force http to https.

**Comment by HirokazuNishi** [comment]
Strange. What PHP & Magento version do you use?
Also which version of our extension?

**Comment by mugua** [comment]
Magento 2.3.5-p1, Php 7.3.x and your latest version...

**Comment by mugua** [comment]
It sames that is the Magento 2.3.5-p1's bug...

**Comment by HirokazuNishi** [comment]
Of course I tested 2.3.5-p1 with both PHP 7.2/7.3. The extension works fine for my env.
Did you test it on vanilla Magento 2.3.5-p1?

**Comment by mugua** [comment]
Can you show the secure attribute in frontend ? I only can show httponly and samesite=none.

**Comment by HirokazuNishi** [comment]
My local has both SameSite and Secure flags on frontend. Of course I'm using 2.3.5-p1 and latest.
<img width="1532" alt="samesite-cookie" src="https://user-images.githubusercontent.com/124136/82313019-c1304680-9a02-11ea-9092-fbad7b4ecc70.png">

**Comment by mugua** [comment]
I found that the magento Marketplace has the same issue.
you can have a look at https://marketplace.magento.com
Thanks.

**Comment by mugua** [comment]
![图片](https://user-images.githubusercontent.com/15046097/82314078-d786d400-99fb-11ea-803b-427beb561e31.png)
It the same as me.

**Comment by HirokazuNishi** [comment]
It's not Magento's marketplace issue. These cookies weren't set SameSite or Secure attribute by their origin. You can only control your website scope cookie. However I don't know why your environment doesn't work as expected. Did you install other extensions?

**Comment by mugua** [comment]
yeah, Codazon Infinit theme, wechatpay and alipay extensions and so on...

**Comment by HirokazuNishi** [comment]
Please test vanilla Magento. Extensions and themes sometimes cause serious conflicts.

**Comment by mugua** [comment]
OK, I'LL TRY IT...

**Comment by mugua** [comment]
I run a clean Magento 2.3.5-p1, It is still haven't secure tag...
![图片](https://user-images.githubusercontent.com/15046097/82320390-c347d480-9a05-11ea-81cc-fcf9e7a48bc7.png)

**Comment by mugua** [comment]
it seems that 2.4-dev still have this problem...
https://github.com/magento/magento2/issues/27879

**Comment by niro08** [comment]
@mugua, I had the same problem. It looks like you have a different "base_url" and "base_secure_url". As a result false is passed to the **setCookieSecure** function in vendor/magento/framework/Session/Config.php file (see screenshot)
![2020-08-17 11 06 25 (1)](https://user-images.githubusercontent.com/6094396/90378996-68f8f700-e07a-11ea-9bf5-5699ca0bf30d.jpg)
